### PR TITLE
fix/Bug_58 Simple form ruined when adding several brands in Advanced Search

### PR DESCRIPTION
**Pull request by elador13**
See - https://app.asana.com/0/1202525679663009/1202928610113602/f

fixed-simple-form

### CODE REVIEW

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx [hunk 1]
```diff
@@ -30,9 +32,21 @@ const SimpleAutoFilter: FC = () => {
 
   const { filters, brandDetails } = useAppSelector((state) => state.carFilter);
 
+  const {
+    id: detailId,
+    brandId,
+    modelIds,
+  } = useMemo(() => brandDetails[0], [brandDetails[0]]);
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: `useMemo` is used to _wrap some calculations or complex evaluation_. And `brandDetails[0], [brandDetails[0]]` is misusing of the `useMemo`

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx [hunk 2]
```diff
@@ -80,7 +90,64 @@ const SimpleAutoFilter: FC = () => {
...
+            id: item.id,
+          } as AutocompleteValueType),
+      ),
+
+    [models],
+  );
+
+  const selectedBrandName = useMemo(() => {
+    return getValueById(brands || [], brandId);
+  }, [brandDetails[0]]);
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: Please extract brandDetails[0] into a variable - `const [smth] = brandDetails;`

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx [hunk 3]
```diff
@@ -80,7 +90,64 @@ const SimpleAutoFilter: FC = () => {
...
+
+  const selectedModelName = useMemo(
+    () => getValueById(models || [], modelIds[0]),
+    [modelIds[0]],
+  );
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: Arrays indexing is hard to read

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx [hunk 4]
```diff
@@ -99,18 +166,32 @@ const SimpleAutoFilter: FC = () => {
...
+              value={selectedModelName}
+              onChange={handleSelectModel}
             />
-          ))}
+          ) : (
+            <SelectField
+              id="disabled"
+              name="Model"
+              value=""
+              disabled={true}
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: `disable={true}` can be shortened just to the `disabled`

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx [hunk 5]
```diff
@@ -99,18 +166,32 @@ const SimpleAutoFilter: FC = () => {
...
+            <AutocompleteInput
+              label="Model"
+              options={modelsOptions}
+              value={selectedModelName}
+              onChange={handleSelectModel}
             />
-          ))}
+          ) : (
+            <SelectField
+              id="disabled"
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: `id`?

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx (hunk 3 above)
Comment (elador13) [reply frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: I simplified where I could. In this case I show only first selected model in the simple form.

File: frontend/src/components/simple-auto-filter/simple-auto-filter.tsx (hunk 2 above)
Comment (elador13) [reply frontend/src/components/simple-auto-filter/simple-auto-filter.tsx]: That's already done on line 35, just forgot use variables in `deps`

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**